### PR TITLE
Replace ipaplatform's symlinks with __path__

### PR DESCRIPTION
**Pull request by tiran**
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipaserver/dcerpc.py [hunk 1]
```diff
@@ -58,7 +58,7 @@
 import pysss_nss_idmap
 import pysss
 import six
-from ipaplatform.paths import paths
+from ipaplatform.paths import paths  # pylint: disable=E0401
```
Comment (tiran) [inline ipaserver/dcerpc.py]: For some reasons I do not understand, this import is not handled correctly by the pylint module extender.

**Comment by pspacek** [comment]
Uh, it is quite a lot of code to get rid of few symlinks. Is it worth? @jcholast @mbasti-rh @martbab @tomaskrizek and others?

File: ipaserver/dcerpc.py (hunk 1 above)
Comment (jcholast) [inline ipaserver/dcerpc.py]: Please use message ids (`import-error`) rather than message codes (`E0401`) in pylint comments.

**Comment by tiran** [comment]
I can also make the ```__init__.py.in``` go away and let everything be handled by auto-detecting the current distribution with Python's platform module.

File: ipaplatform/__init__.py.in [hunk 2]
```diff
@@ -5,8 +5,58 @@
 '''
 Module containing platform-specific functionality for every platform.
 '''
+import sys
 
 NAME = "__PLATFORM__"
```
Comment (jcholast) [inline ipaplatform/__init__.py.in]: I don't see any benefit in deciding which platform module to use at runtime.

File: ipaplatform/__init__.py.in [hunk 3]
```diff
@@ -5,8 +5,58 @@
...
+
+    def get_code(self, fullname):
+        self.load_module(fullname)
+        return None
+
+    def get_source(self, fullname):
+        self.load_module(fullname)
+        return None
+
+sys.meta_path.append(IpaPlatformImporter())
```
Comment (tiran) [inline ipaplatform/__init__.py.in]: The ```__path__``` trick results in a slightly different behavior:

##  ```__path__```

```
>>> import ipaplatform.paths, ipaplatform.fedora.paths
>>> ipaplatform.paths.paths is ipaplatform.fedora.paths.paths
False
```

##  meta importer

```
>>> import ipaplatform.paths, ipaplatform.fedora.paths
>>> ipaplatform.paths.paths is ipaplatform.fedora.paths.paths
True
```

File: ipaplatform/__init__.py.in (hunk 2 above)
Comment (tiran) [inline ipaplatform/__init__.py.in]: I don't see any benefits to hard-code the platform at build time.

File: ipaplatform/__init__.py.in (hunk 3 above)
Comment (jcholast) [inline ipaplatform/__init__.py.in]: The behavior with `__path__` is exactly the same as with the symlinks, which work just fine.

Comment (tiran) [inline ipaplatform/__init__.py.in]: Let's go for ```__path__``` and hope that it won't break in the future. I'm going to ask Brett for his opinion on the trick later, too.

By the way your suggested alternative ```__path__.append('__PLATFORM__')``` does not work. Paths in ```__path__``` must be relative to sys.path elements, not relative to the current package directory. The correct path is ```__path__.append(os.path.join(__name__, NAME))```.

**Comment by pspacek** [comment]
NACK

freeipa-server-4.4.90.201610190959GITf8012c0-0.fc24.x86_64 it broke following call which is used in RPM `%post client` scriptlet:
~~~
$ python2 -c 'from ipapython.certdb import update_ipa_nssdb; update_ipa_nssdb()'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/ipapython/certdb.py", line 28, in <module>
    from ipaplatform.paths import paths
ImportError: No module named paths
~~~

**Comment by dkupka** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/8f98fa1bd5f1da207fab6f89b75e0cdc19d00797

**Pull request closed without merging**